### PR TITLE
test: add V3 test case for inpool->inpackage->inpackage

### PR DESCRIPTION
**Pull request by instagibbs**

### CODE REVIEW

**Comment by instagibbs** [comment]
@glozow current coverage seems to cover inpackage->inpackage->inpackage, so this might be useful

File: test/functional/mempool_accept_v3.py [hunk 1]
```diff
@@ -262,6 +262,16 @@ def test_v3_ancestors_package(self):
...
+        tx_v3_grandchild = self.wallet.create_self_transfer(utxo_to_spend=tx_v3_child["new_utxo"], version=3)
+
+        node.sendrawtransaction(tx_v3_parent["hex"])
+        self.check_mempool([tx_v3_parent["txid"]])
+
+        result = node.submitpackage([tx_v3_child["hex"], tx_v3_grandchild["hex"]])
+        assert_equal(result["package_msg"], f"v3-violation, tx {tx_v3_grandchild['txid']} (wtxid={tx_v3_grandchild['wtxid']}) would have too many ancestors")
+
     @cleanup(extra_args=["-acceptnonstdtxn=1"])
     def test_v3_ancestors_package_and_mempool(self):
```
Comment (tdb3) [inline test/functional/mempool_accept_v3.py:276]: Is this scenario (parent is in mempool, child->grandchild package is submitted and is rejected) at least in part already covered in the `test_v3_ancestors_package_and_mempool()` test?  Maybe I'm overlooking something?

```python
    def test_v3_ancestors_package_and_mempool(self):
        """
        A v3 transaction in a package cannot have 2 v3 parents.
        Test that if we have a transaction graph A -> B -> C, where A, B, C are
        all v3 transactions, that we cannot use submitpackage to get the
        transactions all into the mempool.

        Verify, in particular, that if A is already in the mempool, then
        submitpackage(B, C) will fail.
        """
```

**Review by tdb3 — COMMENTED** [review]
Always good to find new test coverage opportunities.  
Pulled and ran functional test.  Passed.
Inline comment left.

File: test/functional/mempool_accept_v3.py (hunk 1 above)
Comment (instagibbs) [reply test/functional/mempool_accept_v3.py:276]: ya know what, it is. Closing this PR.

**Comment by glozow** [comment]
Agree this case is already covered. Though from offline conversation it sounds like you're hitting some interesting things when it's a package RBF, so will keep this case in mind. Thanks as well @tdb3 :)

**Pull request closed without merging**